cpp/src/slice2cs/Gen.cpp

```diff
@@ -1874,7 +1762,7 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& p)
             true);
         emitObsoleteAttribute(p, _out);
         _out << nl << taskResultType(p, ns);
-        _out << " " << p->name() << "Async" << spar << inParams
+        _out << " " << name << "Async" << spar << inParams
```

Comment:
Don't we nee to remove `@` if present in name?

Reply:
We can but it's optional. `@` creates a syntax error only when it's in the middle of an identifier. It's ignored as prefix.

Reply:
The approach that I took was to only remove the `@` when absolutely necessary and for internal code that users won't be seeing.

I think it's desirable for `cs:identifier` to be respected as much as possible. And if you put `cs:identifier:@throw` we will use `@throw` everywhere, even if the escape character isn't strictly necessary there.

The less magical `cs:identifier` is the better IMO.